File: roscpp_basic_pub/src/helloworld_publisher.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

int main(int argc, char* argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<rclcpp::Node>("helloworld_publisher");
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr helloworld_publisher;
  rclcpp::TimerBase::SharedPtr timer;
  auto qos_profile = rclcpp::QoS(rclcpp::KeepLast(10));
  size_t count;
  helloworld_publisher = node->create_publisher<std_msgs::msg::String>(
    "helloworld", qos_profile);
  timer = node->create_wall_timer(
    1s, [node, helloworld_publisher, &count]()->void
    {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello World2: " + std::to_string(count++);
        RCLCPP_INFO(node->get_logger(), "Published message: '%s'", msg.data.c_str());
        helloworld_publisher->publish(msg);
    }
  );
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: rospy_basic_pub/rospy_basic_pub/helloworld_publisher.py
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile
from std_msgs.msg import String

count = 0

def publish_helloworld_msg(helloworld_publisher):
    global count
    msg = String()
    msg.data = 'Hello World : {0}'.format(count)
    helloworld_publisher.publish(msg)
    Node.get_logger('test').info('Published message : {0}'.format(msg.data))
    count += 1


def main(args=Node):
    rclpy.init(args=args)
    node = Node('helloworld_publisher')
    qos_profile = QoSProfile(depth=10)
    helloworld_publisher = node.create_publisher(String, 'helloworld', qos_profile)
    timer = node.create_timer(1, lambda: publish_helloworld_msg(helloworld_publisher))

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')
    finally:
        node.destroy_node()
        rclpy.shutdown()

    return

if __name__=='__main__':
    main()